### ISSUE TITLE
Refactor Codebase for Improved Organization and Maintainability

### ISSUE THREAD

**Issue by JacquesDelfrate** [HeartWise-AI/DeepCORO_CLIP#10]
# Context
Currently, the repository implements video-report pairs matching using a contrastive loss that is compatible with both single GPU and multi-GPU setups via the torchrun API. The latest pull request merged the multi-GPU solution into the existing codebase, which resulted in some code duplication.

To improve code readability, maintainability, and debugging efficiency, we propose to refactor the codebase—adopting a structure inspired by [this PyTorch example file](https://github.com/pytorch/examples/blob/main/distributed/ddp-tutorial-series/multigpu_torchrun.py) with some necessary adjustments based on the specifics of this project.

# Solution 
The refactor will focus on reorganizing the codebase, introducing modularity, and improving maintainability while preserving existing functionality.
##  Codebase Organization
- [x] Clean `script/` folder
   - [ ] Delete useless training and val script
   - [ ] Having one training script to rule them all
- [x] Dataloader Organization (Migrate dataloader-related logic into dedicated files for clarity)
   - [ ] stats_dataset.py
   - [ ] video_dataset.py
   - [ ] video.py 
- [x] Create a well-organized `utils/` folder with the following files
   - [ ] `losses` (implement or refactor loss functions)
   - [ ] `metrics` (evaluation and training metrics)
   - [ ] `parser`
   - [ ] `scheduler` (learning rate scheduler or others)
   - [x] `video` (utils fonctions related to the videos)
   - [ ] `wandb_logger` 
   - [x] `config` ( with `@dataclass` for easy configuration management and runtime sanity checks) 
   - [ ] `registry` (for a modular, plugin-like architecture for the pipeline components)
## New Architecture
- [x] Leverage a registry pattern to improve modularity:
   - [x] Register loss functions
   - [x] Register models
   - [x] Register training pipelines (trainer or runner logic)
   - [x] Register configuration params (`configs`).
   - [x] Register entire pipeline workflows (to enable running different pipeline configurations or model types cleanly)
 ## Test and validation
 - [x] The refactored training pipeline should behave similarly to the current implementation:
   - [ ]  Ensure training loss still trends down from 2.5 to approximately 2.0 (or equivalent, depending on the testing setup).
   - [x] Log/verify results and compare before and after the refactor to validate functional parity.
